refactor!: nuking Aztec Node proxy methods from PXE

benesjan opened the pull request:
Having these methods exposed on PXE is a relic of a past. PXE JSON RPC server will soon be nuked and we will no longer pass a PXE_URL around and we will just pass NODE_URL.

Note that in some places this results in a worse devex as now it's necessary to pass on the input both AZTEC_NODE_URL and PXE_URL. This is only temporary so would not worry about this.

## Follow-up work
In this PR I don't touch the `PXE` proxy methods that were exposed on `Wallet` as well as it's not yet resolved what to do with those.

I also didn't yet touch `getCurrentBaseFees` as that is used by `Wallet` implementations and that would then force me to pass `AztecNode` on the input of those implementations which is a huge change. For this reason plan on doing that in a PR up the stack.

benesjan commented:
* **#17109** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17109?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17098** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17098?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17098?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on yarn-project/pxe/src/pxe_service/pxe_service.ts:
```diff
@@ -202,61 +194,10 @@ export class PXEService implements PXE {
-    return [result.leafIndex, result.siblingPath];
-  }
-
   public getTxReceipt(txHash: TxHash): Promise<TxReceipt> {
     return this.node.getTxReceipt(txHash);
   }
```
Not removed yet as it's needed by wallet. I think I might anyway drop it from here and have it directly on `Wallet` interface (out of the Pick<PXE, "..."> thing) soon as wallet will most likely end up having access to aztec node and then we can just have it directly call the node instead of going through pxe.

Thunkar commented on yarn-project/cli-wallet/src/cmds/index.ts:
```diff
@@ -455,25 +456,39 @@ export function injectCommands(
     .option('--mint', 'Mint the tokens on L1', false)
     .option('--l1-private-key <string>', 'The private key to the eth account bridging', PRIVATE_KEY)
     .addOption(pxeOption)
```
Should we remove pxeOption?

benesjan replied:
It is still used for now but will be nuked once PXE JSON RPC server is nuked.

Thunkar commented on yarn-project/cli/src/cmds/pxe/block_number.ts:
```diff
@@ -1,9 +1,9 @@
-export async function blockNumber(rpcUrl: string, debugLogger: Logger, log: LogFn) {
-  const client = await createCompatibleClient(rpcUrl, debugLogger);
-  const [latestNum, provenNum] = await Promise.all([client.getBlockNumber(), client.getProvenBlockNumber()]);
+export async function blockNumber(nodeUrl: string, log: LogFn) {
```
This should not be under "pxe"

benesjan replied:
Good catch. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/17098/commits/d2c818bdef55255791719ced10f52c9dd8a86400

Thunkar commented on yarn-project/cli/src/cmds/pxe/get_l1_to_l2_message_witness.ts:
```diff
@@ -1,16 +1,16 @@
 
 export async function getL1ToL2MessageWitness(
-  rpcUrl: string,
+  nodeUrl: string,
```
This shouldn't be under "pxe"

benesjan replied:
https://github.com/AztecProtocol/aztec-packages/pull/17098/commits/d2c818bdef55255791719ced10f52c9dd8a86400

Thunkar commented on yarn-project/cli/src/cmds/pxe/get_logs.ts:
```diff
@@ -9,12 +9,11 @@ export async function getLogs(
   afterLog: LogId,
   contractAddress: AztecAddress,
-  rpcUrl: string,
+  nodeUrl: string,
```
Same, all of these commands should go elsewhere

benesjan replied:
https://github.com/AztecProtocol/aztec-packages/pull/17098/commits/d2c818bdef55255791719ced10f52c9dd8a86400

Thunkar commented on yarn-project/end-to-end/src/e2e_nested_contract/nested_contract_test.ts:
```diff
@@ -39,10 +40,12 @@ export class NestedContractTest {
         [{ address: this.defaultAccountAddress }] = deployedAccounts;
         this.pxe = pxe;
+        this.aztecNode = aztecNode;
+        this.aztecNode = aztecNode;
```
Duplicated

benesjan replied:
https://github.com/AztecProtocol/aztec-packages/pull/17098/commits/bef040baf27a8a35f43d3a9dfbb92f6e5aa3132a

benesjan committed:
refactor!: nuking Aztec Node proxy methods from PXE

Having these methods exposed on PXE is a relic of a past. PXE JSON RPC server will soon be nuked and we will no longer pass a PXE_URL around and we will just pass NODE_URL.

Note that in some places this results in a worse devex as now it's necessary to pass on the input both AZTEC_NODE_URL and PXE_URL. This is only temporary so would not worry about this.

## Follow-up work
In this PR I don't touch the `PXE` proxy methods that were exposed on `Wallet` as well as it's not yet resolved what to do with those.

I also didn't yet touch `getCurrentBaseFees` as that is used by `Wallet` implementations and that would then force me to pass `AztecNode` on the input of those implementations which is a huge change. For this reason plan on doing that in a PR up the stack.
